lib/generator/js.rb

```diff
@@ -271,6 +273,18 @@ def js_for_data_criteria(force_sources=nil)
         HQMF2JS::Generator.render_template('data_criteria', {'all_criteria' => @doc.specific_occurrence_source_data_criteria(force_sources).concat(@doc.all_data_criteria), 'measure_period' => @doc.measure_period})
       end
       
+      def js_for_variables()
+        variables_js = ""
+        variables_js += "hqmfjs.VARIABLES = function(patient, initialSpecificContext) {\n" 
+        @doc.source_data_criteria.each do |criteria|
+          if criteria.type == :derived
+            variables_js += "hqmfjs." + criteria.id + "(patient, initialSpecificContext);\n"
+          end
+        end
+        variables_js += "return false;\n}"
```

Comment:
ish. In the inject logger code, it assumes the existence of a `result` because it calls `result.specificContext` on it. `false` manages this just fine. Returning nothing doesn't. I could add a check for null/undefined in the inject logger code, but since I've never touched this code, I thought it would be better to mod the new code to fit the old than the other way around. 

I could also make the variables call return the AND result of all the given variables, which might be more logical. However, this result doesn't affect processing so I just returned `false`.

Let me know if you think it's worth changing the inject logger code or changing the returned value to something that aligns more with the variable outputs.